### ISSUE TITLE
Support postponed evaluation of annotations in for class variables

### ISSUE THREAD

**Issue by quodlibetor** [PyCQA/pylint#2838]
### Steps to reproduce
Related to #2069 , but this is about class variable annotations introduced with [PEP 526](https://www.python.org/dev/peps/pep-0526/)/Python3.6.

```python
# cat example.py
from __future__ import annotations

class Example:
    obj: Other

class Other:
    ...
```
Pylint is sad:
```
$ pylint example.py
************* Module example
example.py:4:9: E0601: Using variable 'Other' before assignment (used-before-assignment)
```
Mypy is fine:
```
$ mypy example.py
$
```

### Current behavior

Warns about use-before-assignment, even in the presence of `from __future__ import annotations`

### Expected behavior

No error.

### pylint --version output

```
$ pylint --version
pylint 2.4.0-dev0
astroid 2.2.5
Python 3.7.2 (default, Feb  4 2019, 16:07:25)
[Clang 10.0.0 (clang-1000.11.45.5)]

$ git lg -1
* 75a04791 - (HEAD -> master, origin/master, origin/HEAD) ...
```

**Comment by PCManticore** [comment]
Thank you, this makes sense.
